iron-fit-behavior.html

```diff
@@ -253,11 +253,13 @@
       if (typeof this._isRTL === 'undefined') {
         this._isRTL = window.getComputedStyle(this).direction == 'rtl';
       }
-          
       this.positionTarget = this.positionTarget || this._defaultPositionTarget;
       if (this.autoFitOnAttach) {
-        if (window.getComputedStyle(this).display === 'none') {
-          setTimeout(function() {
+        // If invisible or in a shadow root that's still not connected, wait a tick.
+        if (window.getComputedStyle(this).display === 'none' ||
+          document.body.contains(this) === false) {
```

Comment:
`document.body.contains(this)` will only be true when `this` isn't inside any ShadowRoot, even if its connected. Are you finding that `attached` is getting fired when `this` isn't connected? If so, this could be another reactions queue issue.